modules/effects/spec/integration.spec.ts

```diff
@@ -90,6 +92,56 @@ describe('NgRx Effects Integration spec', () => {
     });
   });
 
+  it('should execute user provided root effects', (done: DoneFn) => {
```

Comment:
Instead of writing these tests, what about adding the providers in the `should dispatch init actions in the correct order` spec? This way, we also test the order of effects registration

Reply:
Could we also create a simple unit test to verify the effect is added?
Something as:

```ts
describe('when registered with the USER_PROVIDED_FEATURE_EFFECTS token', () => {
    class SourceA {}
    const sourceA = new SourceA();

    let mockEffectSources: { addEffects: jasmine.Spy };

    beforeEach(() => {
      TestBed.configureTestingModule({
        imports: [EffectsModule.forFeature()],
        providers: [
          SourceA,
          {
            provide: EffectsRootModule,
            useValue: {
              addEffects: jasmine.createSpy('addEffects'),
            },
          },
          {
            provide: USER_PROVIDED_FEATURE_EFFECTS,
            useValue: [SourceA],
            multi: true,
          },
        ],
      });
      mockEffectSources = TestBed.get(EffectsRootModule);
    });

    it('should add effects when provided', () => {
      TestBed.get(EffectsFeatureModule);
      expect(mockEffectSources.addEffects).toHaveBeenCalledWith(sourceA);
    });
  });
```